projects/fw_103_hw/src/adc_converter.cpp

```diff
@@ -0,0 +1,58 @@
+#include "adc.h"
+#include "gpio.h"
+#include "gpio_it.h"
+#include "interrupt.h"
+#include "wait.h"
+#include "log.h"
+
+static void prv_button_press(const GpioAddress *address, void *context) {
+    uint16_t adc_data = 0;
+    adc_read_converted(*read_channel, &adc_data);
+    LOG_DEBUG("%d\n", adc_data);
+}
+
+int main(void) {
+    interrupt_init();
+    soft_timer_init();
+    gpio_init();
+    gpio_it_init();
+
+    GpioAddress read_addr = {.port = GPIO_PORT_A, .pin = 6};
+    GpioAddress button_addr = {.port = GPIO_PORT_B, .pin = 2};
+
+    GpioSettings read_settings = {
+            GPIO_DIR_IN;
+            GPIO_STATE_LOW;
+            GPIO_RES_NONE;
+            GPIO_ALTFN_ANALOG;
+    };
+
+    GpioSettings button_settings = {
+            GPIO_DIR_IN;
+            GPIO_STATE_LOW;
+            GPIO_RES_NONE;
+            GPIO_ALTFN_ANALOG;
+    };
```

Comment:
Using semicolons won't compile here, you need commas instead. Also, please use the GNU struct initialization syntax `.direction = GPIO_DIR_IN`, `.state = GPIO_STATE_LOW`, etc.; it makes it easier to read and decreases the chance that you'll order something wrong and mess up the initialization.